src/main/webapp/app/entities/participation/participation-websocket.service.ts

```diff
@@ -20,7 +21,7 @@ export interface IParticipationWebsocketService {
 
 @Injectable({ providedIn: 'root' })
 export class ParticipationWebsocketService implements IParticipationWebsocketService {
-    cachedParticipations: Map<number /* ID of participation */, Participation> = new Map<number, Participation>();
+    cachedParticipations: Map<number /* ID of participation */, StudentParticipation> = new Map<number, StudentParticipation>();
```

Comment:
This also means we need a new websocket service for the Template/SolutionParticipations.
So this would mean we need a generic ParticipationWebsocketService with a type parameter for the Participation to make reuse possible. This also means that we need different websocket channels for transporting participation messages for the 3 types.

Reply:
Thinking about this again, my comment was incorrect: we don't use the participation caching for template / solution participations, just for receiving new results. So it can stay as is.